### PR TITLE
cmd: don't treat help as an error

### PR DESCRIPTION
**Pull request by zyga**
This makes ```snap foo --help | sensible-pager``` work correctly and also gets rid of the annoying "error: " prefix in the usage string.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: cmd/snap/main.go [hunk 1]
```diff
@@ -45,8 +45,20 @@ func init() {
...
-		fmt.Fprintf(os.Stderr, "error: %v\n", err)
-		os.Exit(1)
+		isHelp := false
+		switch err.(type) {
+		case *flags.Error:
+			if err.(*flags.Error).Type == flags.ErrHelp {
+				isHelp = true
+			}
+		}
+		if isHelp {
```
Comment (niemeyer) [inline cmd/snap/main.go]: if _, ok := err.(*flags.Error); ok {

**Comment by niemeyer** [comment]
LGTM assuming that form is taken.

File: cmd/snap/main.go (hunk 1 above)
Comment (zyga) [inline cmd/snap/main.go]: Ah thanks, I forgot about this form of type assertion, nice! Fixed.

**Comment by chipaca** [comment]
:+1:

**Pull request merged**